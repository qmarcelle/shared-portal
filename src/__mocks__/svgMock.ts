import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
