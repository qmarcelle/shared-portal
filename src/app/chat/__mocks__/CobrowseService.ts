export class CobrowseService {
  constructor() {
    // Empty constructor
  }

  initialize = jest.fn().mockResolvedValue({});
  createSession = jest.fn().mockResolvedValue('test-code');
  endSession = jest.fn().mockResolvedValue({});
}
