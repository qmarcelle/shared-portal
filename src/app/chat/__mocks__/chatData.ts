import { ChatConfig, ClientType, Plan, UserEligibility } from '../models/chat';

/**
 * Mock chat configuration for testing
 */
export const mockChatConfig: ChatConfig = {
  token: 'mock-jwt-token',
  endPoint: 'https://api.example.com/chat',
  demoEndPoint: 'https://demo-api.example.com/chat',
  opsPhone: '800-123-4567',
  memberFirstname: 'John',
  memberLastname: 'Doe',
  memberId: 'MEMBER123',
  groupId: 'GROUP456',
  planId: 'PLAN789',
  planName: 'Premium Health Plan',
  businessHours: {
    isOpen24x7: true,
    days: [
      { day: 'Monday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Tuesday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Wednesday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Thursday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Friday', openTime: '08:00', closeTime: '20:00' },
      { day: 'Saturday', openTime: '09:00', closeTime: '17:00' },
      { day: 'Sunday', openTime: '09:00', closeTime: '17:00' },
    ],
  },
  cobrowseSource: 'https://cobrowse.example.com',
  cobrowseURL: 'https://js.cobrowse.io/CobrowseIO.js',
  coBrowseLicence: 'mock-cobrowse-license-key',
};

/**
 * Mock user eligibility data for testing
 */
export const mockUserEligibility: UserEligibility = {
  isChatEligibleMember: true,
  isDemoMember: false,
  isAmplifyMem: false,
  groupId: 'GROUP456',
  memberClientID: ClientType.Default,
  getGroupType: 'Commercial',
  isBlueEliteGroup: false,
  isMedical: true,
  isDental: true,
  isVision: true,
  isWellnessOnly: false,
  isCobraEligible: false,
  chatHours: 'Monday-Friday: 8:00 AM - 6:00 PM',
  rawChatHours: 'M_F_8_6',
  isChatbotEligible: true,
  memberMedicalPlanID: 'PLAN789',
  isIDCardEligible: true,
  memberDOB: '1980-01-01',
  subscriberID: 'SUB123',
  sfx: '01',
  memberFirstname: 'John',
  memberLastName: 'Doe',
  userID: 'USER456',
  isChatAvailable: true,
  routingchatbotEligible: true,
  idCardChatBotName: 'speechstorm-chatbot',
  // Added fields for enhanced routing
  RoutingChatbotInteractionId: '',
  coverage_eligibility: '',
  lob_group: 'Commercial',
  Origin: 'member-portal',
  Source: 'web',
};

/**
 * Mock BlueCare user eligibility for testing specialized routing
 */
export const mockBlueCareUserEligibility: UserEligibility = {
  ...mockUserEligibility,
  memberClientID: ClientType.BlueCare,
  lob_group: ClientType.BlueCare,
  getGroupType: 'BlueCare',
};

/**
 * Mock Dental-only user eligibility for testing specialized routing
 */
export const mockDentalOnlyUserEligibility: UserEligibility = {
  ...mockUserEligibility,
  isMedical: false,
  isDental: true,
  isVision: false,
  isWellnessOnly: false,
  coverage_eligibility: 'dental_only',
};

/**
 * Mock ID Card request user eligibility for testing specialized routing
 */
export const mockIdCardUserEligibility: UserEligibility = {
  ...mockUserEligibility,
  isIDCardEligible: true,
  RoutingChatbotInteractionId: 'ID_CARD_REQ',
};

/**
 * Mock current plan for testing
 */
export const mockCurrentPlan: Plan = {
  planId: 'PLAN789',
  planName: 'Premium Health Plan',
  isEligibleForChat: true,
  isCurrentPlan: true,
};

/**
 * Mock available plans for testing multi-plan scenarios
 */
export const mockAvailablePlans: Plan[] = [
  {
    planId: 'PLAN789',
    planName: 'Premium Health Plan',
    isEligibleForChat: true,
    isCurrentPlan: true,
  },
  {
    planId: 'PLAN456',
    planName: 'Basic Dental Plan',
    isEligibleForChat: true,
    isCurrentPlan: false,
  },
];
