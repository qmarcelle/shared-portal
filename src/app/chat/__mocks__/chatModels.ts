// Mock models for chat tests

// Client type enum
export enum ClientType {
  BlueCare = 'BC',
  BlueCarePlus = 'DS',
  CoverTN = 'CT',
  CoverKids = 'CK',
  SeniorCare = 'BA',
  Individual = 'INDV',
  BlueElite = 'INDVMX',
  Default = 'Default',
}

// Chat type enum
export enum ChatType {
  BlueCareChat = 'BlueCare_Chat',
  SeniorCareChat = 'SCD_Chat',
  DefaultChat = 'MBAChat',
}

// Basic chat message type
export interface ChatMessage {
  id?: string;
  text: string;
  sender: 'user' | 'agent' | 'bot';
  timestamp?: number;
}

// Chat session information
export interface ChatSession {
  id: string;
  active: boolean;
  agentName?: string;
  messages: ChatMessage[];
}

// Plan information
export interface PlanInfo {
  planId: string;
  planName: string;
  lineOfBusiness: string; // Maps to ClientType
  isEligibleForChat: boolean;
  businessHours?: string;
}
