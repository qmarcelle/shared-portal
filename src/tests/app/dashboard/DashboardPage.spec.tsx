import DashboardPage from '@/app/dashboard/page';
import { mockedAxios } from '@/tests/__mocks__/axios';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = async () => {
  const page = await DashboardPage();
  return render(page);
};

jest.mock('../../../auth', () => ({
  auth: jest.fn(() =>
    Promise.resolve({
      user: {
        currUsr: {
          firstName: 'Chris',
          plan: {
            planName: 'BlueCross BlueShield of Tennessee',
            subId: '123456',
            grpId: '100000',
            memCk: '123456789',
            coverageType: ['Medical', 'Dental', 'Vision'],
          },
        },
      },
    }),
  ),
}));

describe('Dashboard Page', () => {
  it('should render Welcome Banner UI correctly', async () => {
    mockedAxios.get.mockResolvedValueOnce({
      data: {
        groupData: {
          groupID: '100000',
          groupCK: '21908',
          groupName: 'BlueCross BlueShield of Tennessee',
          parentGroupID: '100001',
          subGroupID: '0001',
          subGroupCK: 28951,
          subGroupName: 'BlueCross BlueShield of Tennessee',
          clientID: 'EI',
          policyType: 'INT',
          groupEIN: '620427913',
        },
        networkPrefix: 'QMI',
        subscriberID: '902218823',
        subscriberCK: '91722400',
        subscriberFirstName: 'CHRIS',
        subscriberLastName: 'HALL',
        coverageTypes: [
          {
            productType: 'M',
            coverageLevel: 'A',
            exchange: false,
            indvGroupInd: '',
            pedAdultInd: '',
          },
          {
            productType: 'D',
            coverageLevel: 'A',
            exchange: true,
            indvGroupInd: 'Group',
            pedAdultInd: 'Adult',
          },
          {
            productType: 'V',
            coverageLevel: '*',
            exchange: true,
            indvGroupInd: 'Group',
            pedAdultInd: 'Adult',
          },
        ],
      },
    });
    const component = await renderUI();
    expect(mockedAxios.get).toHaveBeenCalledWith(
      '/api/member/v1/members/byMemberCk/123456789',
    );
    expect(
      screen.getByText('Plan: BlueCross BlueShield of Tennessee'),
    ).toBeVisible();
    expect(screen.getByText('Policies: Medical, Dental, Vision')).toBeVisible();
    expect(screen.getByText('View Plan Details')).toBeVisible();
    expect(component).toMatchSnapshot();
  });
});
