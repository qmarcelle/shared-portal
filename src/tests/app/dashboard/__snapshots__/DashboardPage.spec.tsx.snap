// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page should render Welcome Banner UI correctly 1`] = `undefined`;
